Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift

```diff
@@ -0,0 +1,188 @@
+    var lowerBound: TimeInterval
+    let upperBound: TimeInterval
+
+    init(lowerBound: TimeInterval, upperBound: TimeInterval) {
+        self.lowerBound = lowerBound
+        self.upperBound = upperBound
+    }
+}
+
+@_objcImplementation extension WKSLinearMediaTrack {
+    let localizedDisplayName: String
+
+    init(localizedDisplayName: String) {
+        self.localizedDisplayName = localizedDisplayName
+    }
+}
+
+// MARK: LinearMediaKit Extensions
+
+extension WKSLinearMediaContentMode {
+    init(_ contentMode: ContentMode?) {
+        guard let contentMode else {
+            self = .none
+            return
+        }
+
+        switch contentMode {
+        case .scaleAspectFit:
+            self = .scaleAspectFit
+        case .scaleAspectFill:
+            self = .scaleAspectFill
+        case .scaleToFill:
+            self = .scaleToFill
+        @unknown default:
+            fatalError()
+        }
+    }
+
+    var contentMode: ContentMode? {
+        switch self {
+        case .none:
+            return nil
+        case .scaleAspectFit:
+            return .scaleAspectFit
+        case .scaleAspectFill:
+            return .scaleAspectFill
+        case .scaleToFill:
+            return .scaleToFill
+        @unknown default:
+            fatalError()
+        }
+    }
+
+    static var `default`: WKSLinearMediaContentMode = .init(.default)
+}
+
+extension WKSLinearMediaContentType {
+    var contentType: ContentType? {
+        switch self {
+        case .none:
+            return nil
+        case .immersive:
+            return .immersive
+        case .spatial:
+            return .spatial
+        case .planar:
+            return .planar
+        case .audioOnly:
+            return .audioOnly
+        @unknown default:
+            fatalError()
+        }
+    }
+}
+
+extension WKSLinearMediaPresentationMode {
+    var presentationMode: PresentationMode? {
+        switch self {
+        case .none:
+            return nil
+        case .inline:
+            return .inline
+        case .fullscreen:
+            return .fullscreen
+        case .fullscreenFromInline:
+            return .fullscreenFromInline
+        case .pip:
+            return .pip
+        @unknown default:
+            fatalError()
+        }
+    }
+}
+
+extension WKSLinearMediaViewingMode {
+    init(_ viewingMode: ViewingMode?) {
+        guard let viewingMode else {
+            self = .none
+            return
+        }
+
+        switch viewingMode {
+        case .mono:
+            self = .mono
+        case .stereo:
+            self = .stereo
+        case .immersive:
+            self = .immersive
+        case .spatial:
+            self = .spatial
+        @unknown default:
+            fatalError()
+        }
+    }
+    
+    var viewingMode: ViewingMode? {
+        switch self {
+        case .none:
+            return nil
+        case .mono:
+            return ViewingMode.mono
```

Comment:
does this need the `ViewingMode` qualifier?

Reply:
nope, will fix